src/shared/components/palette/document/write-document-field.component.ts

```diff
@@ -136,35 +164,30 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
       });
     } else {
 
-      this.selectedFile = null;
-      this.uploadedDocument.get('document_url').setValue(null);
-      this.uploadedDocument.get('document_binary_url').setValue(null);
-      this.uploadedDocument.get('document_filename').setValue(null);
-
+      this.setMandatoryDocumentToEmtpyValues();
```

Comment:
I think you can remove `this.setMandatoryDocumentToEmtpyValues();` and the method `setMandatoryDocumentToEmtpyValues()`  and reuse the ones we introduced. Change to:
```if (this.isAMandatoryComponent()) {
       this.selectedFile = null;
        this.setDocumentFormValuesWithValidator(null, null, null);
        this.displayFileErrors();
      }```